smartsim/_core/launcher/launcher.py

```diff
@@ -69,6 +69,10 @@ def run(self, step: Step) -> t.Optional[str]:
     def stop(self, step_name: str) -> StepInfo:
         raise NotImplementedError
 
+    @abc.abstractmethod
+    def cleanup(self) -> None:
+        raise NotImplementedError
```

Comment:
I'm not sure that I like the idea of making this an abstract method if we are going to make it a no-op on most of the launchers.

Maybe we can just make it a hook method w/ a default implementation that is a no-op? When I did a (naive) grep for where this method is being used, I could seem to find it anywhere in `smartsim`; does this method need to exist on the base class at all?

Reply:
Yeah, I'm on the fence. I can see this being used more in the future, but you are right, this is currently only a dragon thing.

Reply:
Which means I made it only a `DragonLauncher` method for now.